[IMP] inventory: thd batch updates (#2556)
odoo/documentation: content/applications/inventory_and_mrp/inventory/routes/concepts/use-routes.rst

  after-sales services, or supplier returns;
- Help rental management by generating automated return moves for rented products.

To configure a route such as one of those above, open the **Inventory Application** and go to
:menuselection:`Configuration --> Settings`. Then, enable the *Multi-Step Routes* feature.

.. image:: use_routes/multi-steps-routes-feature.png
   :align: center
   :alt: Activate the multi-steps feature in Odoo Inventory.

.. note::
   The *Storage Locations* feature is automatically activated with the *Multi-Step Routes*.

Once this first step is completed, you have the choice between pre-configured routes or custom
ones. Both are explained below.

Pre-configured Routes
---------------------

Odoo's pre-configured routes are available in the warehouses' menu. To find it, go to
:menuselection:`Inventory --> Configuration --> Warehouses`. There, open your warehouse and edit it
to see the pre-configured routes for incoming and outgoing shipments.

.. image:: use_routes/example-preconfigured-warehouse.png
   :align: center
   :alt: preconfigured warehouse in Odoo Inventory

> Some more advanced routes, such as pick-pack-ship, are already configured to make your life easier.
> Those are based on your choice for shipments. Once you made your choice, head to
> :menuselection:`Inventory --> Configuration --> Routes` to see the routes Odoo generated for you.

Reviewer: ```suggestion
Some more advanced routes, such as pick-pack-ship, are also available. The user can select the route
that best fits their business needs. Once the :guilabel:`Incoming Shipments` and :guilabel:`Outgoing
Shipments` routes are set, head to :menuselection:`Inventory --> Configuration --> Routes` to see
the specific routes that Odoo generated.
```

.. image:: use_routes/preconfigured-routes.png
   :align: center
   :alt: View of all the preconfigured routes Odoo offers.

> Opening one of those routes allows you to see on what you can apply the route. Here, all the
> selected product categories in the *YourCompany* warehouse are set up to follow the 3-steps route.

Reviewer: ```suggestion
On the :guilabel:`Routes` page, click on a route to open the route form. In the route form, the user
can view which places the route is :guilabel:`Applicable On`. The user can also set the route to
only apply on a specific :guilabel:`Company`. This is useful for multi-company environments; for
example, a user can have a company and warehouse in Country A and a second company and warehouse in
Country B.
```

.. image:: use_routes/routes-example.png
   :align: center
   :alt: View of a route example applicable on category and warehouse.

> Easier for many businesses, this process might not fit yours. Thus, you can configure custom routes
> in which you can define your own rules, but also the source and destination location of each
> action. As an example, here are the rules for the pre-configured route.

Reviewer: ```suggestion
At the bottom of the route form, the user can view the specific :guilabel:`Rules` for the route.
Each :guilabel:`Rule` has an :guilabel:`Action`, a :guilabel:`Source Location`, and a
:guilabel:`Destination Location`.
```

.. image:: use_routes/operations-on-transfers.png
   :align: center
   :alt: View of the operations created by a pull from transfer.

.. note::
   If the source document is the same sale order, the status is not the same. In fact, the status
   is **Waiting Another Operation** if the previous transfer in the list is not done yet.

.. image:: use_routes/waiting-status.png
   :align: center
   :alt: View of the transfers various status at the beginning of the process.

To prepare the order, conveyors need products packed at the output area, so an internal transfer is
requested from the packing zone.

.. image:: use_routes/detailed-operations-2.png
   :align: center
   :alt: View of the detailed operations for a transfer between the packing and output zones.

Obviously, the packing zone needs products ready to be packed. So, an internal transfer is
requested to the stock and employees can gather the required products from the warehouse.

.. image:: use_routes/detailed-operations-transfer.png
   :align: center
   :alt: View of the detailed operations for a transfer between the stock and packing zones.

> As explained in this documentation introduction, the last step in the process (here the delivery
> order) is the first to be triggered, which then triggers other rules until we reach the first step
> in the process (here, the internal transfer from to stock to the packing area). Now, everything is
> ready to be processed so the customer can get the ordered items.

Reviewer: ```suggestion
As explained in the introduction of the documentation, the last step in the process (for this route,
the delivery order) is the first to be triggered, which then triggers other rules until we reach the
first step in the process (here, the internal transfer from the stock to the packing area). Now,
everything is ready to be processed so the customer can get the ordered items.
```

> In our case, the product is delivered to the customer when all the rules have been triggered and
> the transfers done.

Reviewer: ```suggestion
In this example, the product is delivered to the customer when all the rules have been triggered and
the transfers are done.
```

.. image:: use_routes/transfers-status.png
   :align: center

>    :alt: View of the transfers' status when all the process is done.

Reviewer: ```suggestion
   :alt: View of the transfers' statuses when the route is completed.
```
